Source/common/SNTRule.m

```diff
@@ -48,6 +49,57 @@ - (instancetype)initWithIdentifier:(NSString *)identifier
   return self;
 }
 
+// Converts rule information downloaded from the server into a SNTRule.  Because any information
+// not recorded by SNTRule is thrown away here, this method is also responsible for dealing with
+// the extra bundle rule information (bundle_hash & rule_count).
+- (instancetype)initWithDictionary:(NSDictionary *)dict {
+  if (![dict isKindOfClass:[NSDictionary class]]) return nil;
+
+  self = [super init];
+  if (self) {
+    _identifier = dict[kRuleIdentifier];
+    if (_identifier == nil) {
```

Comment:
I think we should be strict here and require that the dictionary has the correct keys, returning nil otherwise.

Assuming a SHA256 identifier seems problematic since we also take non-hash value identifiers like Team IDs.

Reply:
> I think we should be strict here and require that the dictionary has the correct keys, returning nil otherwise.

I've added a second check that identifier is valid, this was otherwise already the case.

> Assuming a SHA256 identifier seems problematic since we also take non-hash value identifiers like Team IDs.

There are 2 acceptable keys when creating rules: `identifier` and `sha256` - the latter was originally the only valid key, we added `identifier` when team ID rules were added. This condition is for backward compatibility with the old key.